shared/src/main/scala/doodle/core/DrawingContext.scala

```diff
@@ -13,14 +13,18 @@ final case class DrawingContext(
   lineJoin: Option[Line.Join],
 
   fillColor: Option[Color],
+  fillGradient: Option[Gradient],
```

Comment:
I think it would be clearer to replace `fillColor` and `fillGradient` with a single `fill` attribute. Setting both really serves no purpose (gradient overrides plain color fill in that case). Concretely, remove `fillColor` and `fillGradient`, replace with `fill: Option[Fill]` and remove the method `fill` below.